crates/core/src/operations/transaction/mod.rs

```diff
@@ -429,25 +458,96 @@ impl<'a> PreCommit<'a> {
                 table_data: this.table_data,
                 max_retries: this.max_retries,
                 data: this.data,
+                post_commit: this.post_commit_hook.map(|v| PostCommit::from(v)),
             })
         })
     }
 }
 
+/// Represents items for the post commit hook
+pub struct PostCommit {
+    create_checkpoint: bool,
+}
+
+impl From<PostCommitHookProperties> for PostCommit {
+    fn from(value: PostCommitHookProperties) -> Self {
+        PostCommit {
+            create_checkpoint: value.create_checkpoint,
+        }
+    }
+}
+
+impl<'a> PostCommit {
+    async fn maybe_create_checkpoint(
+        &self,
+        table: &Option<&'a dyn TableReference>,
+        log_store: &LogStoreRef,
+        version: i64,
+        commit_data: &CommitData,
+    ) -> DeltaResult<()> {
+        if self.create_checkpoint {
+            if let Some(table) = table {
+                let last_checkpoint =
+                    read_last_checkpoint(log_store.object_store().as_ref(), log_store.log_path())
+                        .await?;
+
+                let checkpoint_interval = table.config().checkpoint_interval() as i64;
+
+                if let Some(last_checkpoint) = last_checkpoint {
+                    if (version - last_checkpoint.version) >= checkpoint_interval {
```

Comment:
Suppose you have multiple writers appending to the table each second and the time it takes for the checkpoint to be written is 5 seconds. Because there is a lag for the checkpoint being visible the writers will create checkpoints for their version at the same time. The modulo logic will prevent this for occuring.

Reply:
Right.. didn't think about that one. I'll switch to using modulo then .

Reply:
@Blajda do you have any preference on the design? This PR or the V2 PR?